File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Action/Utility/SFXAction.ts
import { SoundMaster } from "../../Audio/SoundMaster";
import AbstractGameAction from "../AbstractGameAction";

export class SFXAction extends AbstractGameAction{
    public constructor(public path:string, public pitchVar?:number, public pitchAdjust?:boolean) {
        super();
    }

    async Update(): Promise<boolean> {
        if(this.pitchAdjust)
            SoundMaster.PlayVoiceA(this.path,this.pitchVar);
        else
            SoundMaster.PlayVoice(this.path,this.pitchVar);
        return super.Update();
    }
} 

File: Unity/Assets/Scripts/ModTheSpire/Action/Utility/SFXAction.cs
using Cysharp.Threading.Tasks;
using ModTheSpire.Event;

namespace ModTheSpire.Effect.Other
{
    public class SFXAction : IGameAction
    {
        public float Duration { get; set; }
        public ActionType Type { get; } = ActionType.WAIT;
        private SoundMaster Key;
        private float PitchVar = 0.0f;
        private bool Adjust = false;

        public SFXAction(SoundMaster key, float pitchVar = 0, bool pitchAdjust = false) {
            this.Key = key;
            this.PitchVar = pitchVar;
            this.Adjust = pitchAdjust;
        }

        public void Init()
        {
            throw new System.NotImplementedException();
        }

        public UniTask<bool> Update()
        {
            if (!this.Adjust)
            {
                SoundMaster.PlayVoice(Key, PitchVar);
            }
            else
            {
                SoundMaster.PlayVoice(Key, PitchVar, false);
            }
            return UniTask.FromResult(true);
        }
    }
}